specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2024-04-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,1822 @@
+        "clientCertificatePassword": {
+          "description": "The password for the certificate used to authenticate a Service Principal ",
+          "type": "string",
+          "x-nullable": true,
+          "x-ms-secret": true
+        },
+        "clientCertificateSendChain": {
+          "description": "Specifies whether to include x5c header in client claims when acquiring a token to enable subject name / issuer based authentication for the Client Certificate",
+          "type": "boolean"
+        }
+      }
+    },
+    "ManagedIdentityDefinition": {
+      "description": "Parameters to authenticate using a Managed Identity.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "clientId": {
+          "description": "The client Id for authenticating a Managed Identity.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "ManagedIdentityPatchDefinition": {
+      "description": "Parameters to authenticate using a Managed Identity.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "clientId": {
+          "description": "The client Id for authenticating a Managed Identity.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "OCIRepositoryDefinition": {
+      "description": "Parameters to reconcile to the OCIRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration OCI repository.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster OCI repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster OCI repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "repositoryRef": {
+          "description": "The source reference for the OCIRepository object.",
+          "$ref": "#/definitions/OCIRepositoryRefDefinition"
+        },
+        "layerSelector": {
+          "description": "The layer to be pulled from the OCI artifact.",
+          "$ref": "#/definitions/LayerSelectorDefinition"
+        },
+        "verify": {
+          "description": "Verification of the authenticity of an OCI Artifact.",
+          "$ref": "#/definitions/VerifyDefinition"
+        },
+        "insecure": {
+          "description": "Specify whether to allow connecting to a non-TLS HTTP container registry.",
+          "type": "boolean",
+          "default": false
+        },
+        "useWorkloadIdentity": {
```

Comment:
>useWorkloadIdentity [](http://example.com/codeflow?start=9&length=19)

Consider enum: authenticationType: { WorkloadIdentity, None }

Reply:
We only support workloadIdentity. And we use similar parameter in flux extension and want to keep it consistent with in flux configuration.